### ISSUE TITLE
Configuration files need a "Do not modify managed by Puppet" header

### ISSUE THREAD

**Issue by bschonec** [kakwa/puppet-samba#60]
There's no indication that smb.conf (or any other file) is being managed by Puppet.  I tried prepending a header onto $smbconffile but the method I used wasn't idempotent and caused the $smbconffile to be regenerated on every run of Puppet.

I'm a bit confused as to how the $smbconffile is being build.  Typically, I expect 'concat' from stdlib but it looks like you've created some custom ruby code located in ../lib/ to modify the $smbconffile file.

**Comment by kakwa** [comment]
Yes, this module doesn't use a template, it manages each parameter individually.

It's in fact code derived from inifile module from puppetlabs: https://github.com/puppetlabs/puppetlabs-inifile

Basically, it adds a resource type "samba parameter" where you specify **[section]**, **key** = **value**.

The reason it's done that way is to provide complete flexibility on the content of the configuration file. It's specially useful to have several class using the same file, creating new shares, and supporting arbitrary parameters not handled be default in the module.

Normally it should preserve the comments inside the configuration file, so adding a "DO NOT MODIFY BY HAND" on the first line of the file should be possible.

**Comment by bschonec** [comment]
Any idea on how to do that?  I tried using a template and a plain ole file resource but the smb.conf file gets regenerated after every puppet run.  

Also, this needs to be the first section of the file.  Advice on how to make that happen would be appreciated.

**Comment by kakwa** [comment]
I would try something like that:

https://stackoverflow.com/questions/48706995/add-text-to-the-first-line-of-a-file-with-puppet

On paper, my module should not remove the comment
